libmscore/repeatlist.cpp

```diff
@@ -1,423 +1,480 @@
-                        loop        = 1;
-                        m = jumpToStartRepeat(m);
-                        continue;
-                        }
-                  }
-            if (doJump && !isGoto) {
-                  Jump* s = 0;
-                  foreach(Element* e, m->el()) {
-                        if (e->type() == Element::Type::JUMP) {
-                              s = static_cast<Jump*>(e);
-                              break;
-                              }
-                        }
-                  // jump only once
-                  if (jumps.contains(s)) {
-                        m = m->nextMeasure();
-                        if (endRepeat == _score->searchLabel(s->playUntil()))
-                              endRepeat = 0;
-                        continue;
-                        }
-                  jumps.append(s);
-                  if (s) {
-                        Measure* nm = _score->searchLabel(s->jumpTo());
-                        endRepeat   = _score->searchLabel(s->playUntil());
-                        continueAt  = _score->searchLabel(s->continueAt());
-
-                        if (nm && endRepeat) {
-                              isGoto      = true;
-                              rs->len = m->endTick() - rs->tick;
-                              append(rs);
-                              rs = new RepeatSegment;
-                              rs->tick  = nm->tick();
-                              m = nm;
-                              continue;
-                              }
-                        }
-                  else
-                        qDebug("Jump not found");
-                  }
-            m = m->nextMeasure();
-            }
-
-      if (rs) {
-            Measure* lm = _score->lastMeasure();
-            rs->len     = lm->endTick() - rs->tick;
-            if (rs->len)
-                  append(rs);
-            else
-                  delete rs;
-            }
-      update();
-      dump();
-      }
-
-//---------------------------------------------------------
-//   jumpToStartRepeat
-//---------------------------------------------------------
-
-Measure* RepeatList::jumpToStartRepeat(Measure* m)
-      {
-      Measure* nm;
-      //
-      // go back to start of repeat or section break
-      // handle special case of end repeat (Measure m) has a section break
-      //
-      for (nm = m; nm && nm != _score->firstMeasure(); nm = nm->prevMeasure()) {
-            if (nm->repeatFlags() & Repeat::START || (nm->sectionBreak() && m != nm)) {
-                  if (nm->sectionBreak() && nm->nextMeasure())
-                        nm = nm->nextMeasure();
-                  break;
-                  }
-            }
-      rs->len = m->tick() + m->ticks() - rs->tick;
-      append(rs);
-
-      rs        = new RepeatSegment;
-      rs->tick  = nm->tick();
-      return nm;
-      }
-
-}
-
+//=============================================================================

```

Comment:
I'm confussed why this diff is doing the whole file...I only really changed unwind and unwindSection